src/fev2xml.cpp

```diff
@@ -0,0 +1,98 @@
+/* xoreos-tools - Tools to help with xoreos development
+ *
+ * xoreos-tools is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos-tools is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos-tools is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos-tools. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Tool to convert TLK files into XML.
+ */
+
+#include <cstring>
+#include <cstdio>
+
+#include "src/version/version.h"
+
+#include "src/common/scopedptr.h"
+#include "src/common/ustring.h"
+#include "src/common/util.h"
+#include "src/common/error.h"
+#include "src/common/platform.h"
+#include "src/common/readfile.h"
+#include "src/common/writefile.h"
+#include "src/common/cli.h"
+
+#include "src/aurora/types.h"
+
+#include "src/xml/fevdumper.h"
+
+#include "src/util.h"
+
+bool parseCommandLine(const std::vector<Common::UString> &argv, int &returnValue,
+					  Common::UString &inFile, Common::UString &outFile);
+
+void dumpFEV(const Common::UString &inFile, const Common::UString &outFile);
+
+int main(int argc, char **argv) {
+	initPlatform();
+
+	try {
+		std::vector<Common::UString> args;
+		Common::Platform::getParameters(argc, argv, args);
+
+		int returnValue = 1;
+		Common::UString inFile, outFile;
+
+		if (!parseCommandLine(args, returnValue, inFile, outFile))
+			return returnValue;
+
+		dumpFEV(inFile, outFile);
+	} catch (...) {
+		Common::exceptionDispatcherError();
+	}
+
+	return 0;
+}
+
+bool parseCommandLine(const std::vector<Common::UString> &argv, int &returnValue,
+					  Common::UString &inFile, Common::UString &outFile) {
+	using Common::CLI::NoOption;
+	using Common::CLI::Parser;
+	using Common::CLI::ValGetter;
+	using Common::CLI::makeEndArgs;
+
+	NoOption inFileOpt(false, new ValGetter<Common::UString &>(inFile, "input file"));
+	NoOption outFileOpt(true, new ValGetter<Common::UString &>(outFile, "output file"));
+	Parser parser(argv[0], "FMOD FEV to XML converter",
+				  "If no output file is given, the output is written to stdout.\n",
+				  returnValue,
+				  makeEndArgs(&inFileOpt, &outFileOpt));
+
+	return parser.process(argv);
+}
+
+void dumpFEV(const Common::UString &inFile, const Common::UString &outFile) {
+	Common::ScopedPtr<Common::SeekableReadStream> fev(new Common::ReadFile(inFile));
+	Common::ScopedPtr<Common::WriteStream> out(openFileOrStdOut(outFile));
+
+	XML::FEVDumper::dump(*out, fev.release());
```

Comment:
`FEVDumper::dump()` doesn't delete the fev, so this will leak. In fact, `FEVDumper::dump()` doesn't even need to take a pointer, it can just take the reference, and no release is necessary here.